frontend-react/src/TelemetryService.test.ts

```diff
@@ -0,0 +1,325 @@
+                ai.initialize();
+
+                expect(console.warn).not.toHaveBeenCalled();
+            });
+        });
+    });
+
+    describe("#getAppInsights", () => {
+        describe("when AppInsights has not been initialized", () => {
+            test("returns null", () => {
+                expect(getAppInsights()).toBeNull();
+            });
+        });
+
+        describe("when AppInsights has been initialized", () => {
+            beforeEach(() => {
+                process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                    "test";
+
+                ai.initialize();
+            });
+
+            afterEach(() => {
+                process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                    oldConnectionString;
+            });
+
+            test("returns the AppInsights instance", () => {
+                expect(getAppInsights()).not.toBeNull();
+            });
+        });
+    });
+
+    describe("#getAppInsightsHeaders", () => {
+        describe("when AppInsights has not been initialized", () => {
+            test("returns an object with an empty string for the session header", () => {
+                expect(getAppInsightsHeaders()).toEqual({
+                    "x-ms-session-id": "",
+                });
+            });
+        });
+
+        describe("when AppInsights has been initialized", () => {
+            beforeEach(() => {
+                process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                    "test-connection-string";
+
+                ai.initialize();
+            });
+
+            afterEach(() => {
+                process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                    oldConnectionString;
+            });
+
+            test("returns an object with the correct value for the session header", () => {
+                expect(getAppInsightsHeaders()).toEqual({
+                    "x-ms-session-id": "test-session-id",
+                });
+            });
+        });
+    });
+
+    describe("#withInsights", () => {
+        let appInsights: ApplicationInsights | null;
+
+        beforeAll(() => {
+            process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                "test-connection-string";
+
+            jest.spyOn(console, "log").mockImplementation(jest.fn);
+            jest.spyOn(console, "info").mockImplementation(jest.fn);
+            jest.spyOn(console, "warn").mockImplementation(jest.fn);
+            jest.spyOn(console, "error").mockImplementation(jest.fn);
+
+            ai.initialize();
+            appInsights = getAppInsights();
+            withInsights(console);
+        });
+
+        afterAll(() => {
+            jest.resetAllMocks();
+
+            process.env.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING =
+                oldConnectionString;
+        });
+
+        describe("when calling console.info", () => {
```

Comment:
I can use `describe.each`/`test.each` to reduce some redundancy here, but I generally try not to over-optimize in tests to make them as readable/updatable as possible.  I can update this if it's an issue, though!